runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/internal/ReconfigureIT.java

```diff
@@ -0,0 +1,139 @@
+        .configurationRoot("io/aklivity/zilla/runtime/engine/internal")
+        .external("app0")
+        .external("app1")
+        .clean();
+
+    @Rule
+    public final TestRule chain = outerRule(engine).around(k3po).around(timeout);
+
+    @After
+    public void cleanupFileSystem() throws Exception
+    {
+        String source = new File(ReconfigureIT.class
+            .getResource("zilla.reconfigure.original.json")
+            .toURI()).getAbsolutePath();
+        String target = new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.json").getAbsolutePath();
+        FileUtils.copy(source, target);
+        new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.missing.json").delete();
+    }
+
+    @Test
+    @Configuration("zilla.reconfigure.json")
+    @Specification({
+        "${app}/client.sent.data.reconfigure.modify/server",
+        "${net}/client.sent.data.reconfigure.modify/client"
+    })
+    public void shouldConnectToZillaOnNewPortWhenConfigModified() throws Exception
+    {
+        k3po.start();
+
+        String source = new File(ReconfigureIT.class
+            .getResource("zilla.reconfigure.after.json")
+            .toURI()).getAbsolutePath();
+        String target = new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.json").getAbsolutePath();
+        FileUtils.copy(source, target);
+        Thread.sleep(2000);
+
+        k3po.notifyBarrier("CONFIG_CHANGED");
+
+        k3po.finish();
+    }
+
+
+    @Test
+    @Configuration("zilla.reconfigure.missing.json")
+    @Specification({
+        "${app}/client.sent.data.reconfigure.create/server",
+        "${net}/client.sent.data.reconfigure.create/client"
+    })
+    public void shouldConnectToZillaWhenConfigCreated() throws Exception
+    {
+        k3po.start();
+
+        String source = new File(ReconfigureIT.class
+            .getResource("zilla.reconfigure.original.json")
+            .toURI()).getAbsolutePath();
+        File configFile = new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.missing.json");
+        String target = configFile.getAbsolutePath();
+        FileUtils.copy(source, target);
+        Thread.sleep(2000);
+
+        k3po.notifyBarrier("CONFIG_CREATED");
+
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("zilla.reconfigure.json")
+    @Specification({
+        "${app}/client.sent.data.reconfigure.delete/server",
+        "${net}/client.sent.data.reconfigure.delete/client"
+    })
+    public void shouldNotConnectToZillaWhenConfigDeleted() throws Exception
+    {
+        k3po.start();
+        File configFile = new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.json");
+        configFile.delete();
+        Thread.sleep(2000);
```

Comment:
In general, it's better not to use `Thread.sleep(delay)` in the tests if it is avoidable as the sleep delays are magic values that work for some platforms but not others and end up being extra large to be conservative, slowing down the build.

Perhaps we can use the WatchService in the test to notice the event, and that is our indication that the engine has also been notified via WatchService? If this works, then the tests would be event-driven and always execute at full speed on all platforms.